crates/cli/src/result_formatting.rs

```diff
@@ -76,7 +76,7 @@ fn print_all_done(item: &AllDone, f: &mut fmt::Formatter<'_>) -> fmt::Result {
 }
 
 fn print_error_log(log: &AnalysisLog, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-    let file_prefix = if log.file == DEFAULT_FILE_NAME {
+    let file_prefix = if log.file.is_empty() || log.file == DEFAULT_FILE_NAME {
```

Comment:
The new `Parser` trait accepts an optional `Path` for the filename, so it doesn't need to depend on the `DEFAULT_FILE_NAME` constant. When no path is given, this will end up as an empty filename in the logs.